src/Std/Time/DateTime/PlainDateTime.lean

```diff
@@ -0,0 +1,369 @@
+  addMinutes dt (-minutes)
+
+/--
+Adds a `Second.Offset` to a `PlainDateTime`, adjusting the minute, hour, and date if the seconds overflow.
+-/
+@[inline]
+def addSeconds (dt : PlainDateTime) (seconds : Second.Offset) : PlainDateTime :=
+  let totalSeconds := dt.time.toSeconds + seconds
+  let days := totalSeconds.ediv 86400
+  let newTime := dt.time.addSeconds seconds
+  { dt with date := dt.date.addDays days, time := newTime }
+
+/--
+Subtracts a `Second.Offset` from a `PlainDateTime`, adjusting the minute, hour, and date if the seconds underflow.
+-/
+@[inline]
+def subSeconds (dt : PlainDateTime) (seconds : Second.Offset) : PlainDateTime :=
+  addSeconds dt (-seconds)
+
+/--
+Adds a `Nanosecond.Offset` to a `PlainDateTime`, adjusting the seconds, minutes, hours, and date if the nanoseconds overflow.
+-/
+@[inline]
+def addNanoseconds (dt : PlainDateTime) (nanos : Nanosecond.Offset) : PlainDateTime :=
+  let nano := Nanosecond.Offset.ofInt dt.time.nano.val
+  let totalNanos := nano + nanos
+  let extraSeconds := totalNanos.ediv 1000000000
+  let nano := Bounded.LE.byEmod totalNanos.val 1000000000 (by decide)
+  let newTime := dt.time.addSeconds extraSeconds
+  { dt with time := { newTime with nano } }
+
+/--
+Subtracts a `Nanosecond.Offset` from a `PlainDateTime`, adjusting the seconds, minutes, hours, and date if the nanoseconds underflow.
+-/
+@[inline]
+def subNanoseconds (dt : PlainDateTime) (nanos : Nanosecond.Offset) : PlainDateTime :=
+  addNanoseconds dt (-nanos)
+
+/--
+Getter for the `Year` inside of a `PlainDateTime`.
+-/
+@[inline]
+def year (dt : PlainDateTime) : Year.Offset :=
+  dt.date.year
+
+/--
+Getter for the `Month` inside of a `PlainDateTime`.
+-/
+@[inline]
+def month (dt : PlainDateTime) : Month.Ordinal :=
+  dt.date.month
+
+/--
+Getter for the `Day` inside of a `PlainDateTime`.
+-/
+@[inline]
+def day (dt : PlainDateTime) : Day.Ordinal :=
+  dt.date.day
+
+/--
+Getter for the `Hour` inside of a `PlainDateTime`.
+-/
+@[inline]
+def hour (dt : PlainDateTime) : Hour.Ordinal dt.time.hour.fst :=
+  dt.time.hour.snd
+
+/--
+Getter for the `Minute` inside of a `PlainDateTime`.
+-/
+@[inline]
+def minute (dt : PlainDateTime) : Minute.Ordinal :=
+  dt.time.minute
+
+/--
+Getter for the `Second` inside of a `PlainDateTime`.
+-/
+@[inline]
+def second (dt : PlainDateTime) : Second.Ordinal dt.time.second.fst :=
+  dt.time.second.snd
+
+/--
+Determines the era of the given `PlainDateTime` based on its year.
+-/
+def era (date : PlainDateTime) : Year.Era :=
+  if date.year.toInt ≥ 0 then
+    .ce
+  else
+    .bce
+
+/--
+Checks if the `PlainDateTime` is in a leap year.
+-/
+def inLeapYear (date : PlainDateTime) : Bool :=
+  date.year.isLeap
+
+/--
+Getter for the `Second` inside of a `PlainDateTime`.
+-/
+@[inline]
+def nanosecond (dt : PlainDateTime) : Nanosecond.Ordinal :=
+  dt.time.nano
+
+instance : HAdd PlainDateTime Day.Offset PlainDateTime where
```

Comment:
Missing `HAdd PlainDateTime Month.Offset PlainDateTime` or did you leave it out so that the user explicitly has to consider whether to clip or roll over?

Reply:
I leave it out so the user needs to decide if it will clip or roll over.